lib/ui/compositing/scene_builder.h

```diff
@@ -109,15 +106,16 @@ class SceneBuilder : public RefCountedDartWrappable<SceneBuilder> {
  private:
   SceneBuilder();
 
-  std::shared_ptr<flutter::ContainerLayer> root_layer_;
-  flutter::ContainerLayer* current_layer_ = nullptr;
+  bool has_layer() const { return !layer_stack_.empty(); };
+  void AddLayer(std::shared_ptr<Layer> layer);
+  void PushLayer(std::shared_ptr<ContainerLayer> layer);
+  void PopLayer();
 
+  std::vector<std::shared_ptr<ContainerLayer>> layer_stack_;
```

Comment:
Is there any particular reason why `std::vector` is used here instead of `std::stack`?

Reply:
std::stack uses std::deque by default internally.  That is not as efficient storage-wise or speed-wise as vector.  You can do std::stack<T, std::vector<T>> but I figured it was just less verbose to use raw vector at that point.  Can change if you want

Reply:
Using vector sounds good. I'm just curious because we were including <stack> but never using it :)

Reply:
Using vector sounds good. I'm just curious because we were including <stack> but never using it :)